### ISSUE TITLE
Can't access property of interface to reference type in another interface

### ISSUE THREAD

**Issue by donaldpipowitch** [Microsoft/TypeScript#10588]
**TypeScript Version:**  1.8.0 (Playground) / 2.1.0-dev

**Code**

```ts
interface Foo {
  bar: string;
}

interface Box<T> {
  container: T;
}

interface FooWithBoxedProps {
  bar: Box<Foo.bar>;
}
```

**Expected behavior:**

Behave exactly like this:

```ts
interface Foo {
  bar: string;
}

interface Box<T> {
  container: T;
}

interface FooWithBoxedProps {
  bar: Box<string>;
}
```

**Actual behavior:** 

`Cannot find namespace 'Foo'.`

**Comment by OlegDokuka** [comment]
It is unreachable because Foo is not a name space, and variable `bar` is a part of instance of type `Foo`.  In typescript philosophy interface is virtual construction and you cant deal with it in any form except extending and inheritance.
So, you cant deal with Foo as with namespace.

Also, you can try something like above to solve your problem

```typescript
interface Foo {
  bar: string;
}

namespace Foo {
 export type bar= string;
}

interface Box<T> {
  container: T;
}

interface FooWithBoxedProps {
  bar: Box<Foo.bar>;
}
```

**Comment by mhegazy** [comment]
duplicate of https://github.com/Microsoft/TypeScript/issues/4640

**Comment by donaldpipowitch** [comment]
Which is closed in favor of #6606. I'll post my use case there.
